base/file.jl

```diff
@@ -273,29 +273,33 @@ Stacktrace:
 function rm(path::AbstractString; force::Bool=false, recursive::Bool=false)
     if islink(path) || !isdir(path)
         try
-            @static if Sys.iswindows()
-                # is writable on windows actually means "is deletable"
-                st = lstat(path)
-                if ispath(st) && (filemode(st) & 0o222) == 0
-                    chmod(path, 0o777)
-                end
-            end
             unlink(path)
         catch err
-            if force && isa(err, IOError) && err.code==Base.UV_ENOENT
-                return
+            if isa(err, IOError)
+                force && err.code==Base.UV_ENOENT && return
+                @static if Sys.iswindows()
+                    if err.code==Base.UV_EACCES && endswith(path, ".dll")
+                        # Loaded DLLs cannot be deleted on Windows, even with posix delete mode
+                        # but they can be moved. So move out to allow the dir to be deleted
+                        # TODO: Add a mechanism to delete these moved files after dlclose or process exit
+                        dir = mkpath(joinpath(tempdir(), "julia_delayed_deletes"))
```

Comment:
I think adding something to `Pkg.gc()` that checks for this folder and attempts to delete everything inside (silently failing if something is still opened) should work well.  Moving it outside of the depot is good in once sense, as it allows the depot to be deleted, but it's bad in another sense in that it's more likely to be on a different disk, which will probably fail the `mv()` as I doubt you can move a mmap'ed file across device boundaries.   That's probably quite an edge case however.

Reply:
Yeah sounds good https://github.com/JuliaLang/Pkg.jl/pull/3812